packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,137 @@
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/_types/types";
+import { SmartAccount } from "permissionless/accounts";
+import {
+    Abi,
+    Address,
+    BaseError,
+    Chain,
+    ContractFunctionRevertedError,
+    EstimateFeesPerGasReturnType,
+    Hex,
+    PublicClient,
+    TransactionReceipt,
+    Transport,
+    WaitForTransactionReceiptTimeoutError,
+} from "viem";
+
+import { CrossmintErrors, CrossmintSDKError } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+};
+
+export type TransactionServiceResendCallback = (resendAttempts: number) => Promise<{ resend: boolean }>;
+
+export class SendTransactionError extends CrossmintSDKError {
+    constructor(message: string, public readonly viemError: BaseError) {
+        super(message, CrossmintErrors.SEND_TRANSACTION_FAILED);
+    }
+}
+
+export class SendTransactionExecutionRevertedError extends CrossmintSDKError {
+    constructor(
+        message: string,
+        public readonly viemError: BaseError,
+        public readonly revertError: ContractFunctionRevertedError,
+        public txId: string | undefined
+    ) {
+        super(message, CrossmintErrors.SEND_TRANSACTION_FAILED);
+    }
+}
```

Comment:
The reason this error configuration was chosen is that there are others reasons that a transaction can't be sent, e.g.:
- Out of funds
- Nonce issues
- Gas issues

This approach allows us to surface those errors while also:
- Making it easy to handle execution reverted errors specifically
- Surfacing full information for execution reverted errors

Reply:
> Out of funds
> Nonce issues
> Gas issues

How do these look? Are these SendTransactionErrors or specific subclasses? I think we should document all in the TSDoc of send transaction if they have different handlings

Reply:
See https://github.com/Crossmint/crossmint-sdk/pull/665/files#r1717524133. We can't hope to write custom errors for everything that can go wrong when sending a transaction, so we include the viem error in any error thrown by this function while also including subclasses that represent common failure scenarios